tsdb/engine/tsm1/compact.go

```diff
@@ -661,6 +661,19 @@ func (c *DefaultPlanner) Release(groups []CompactionGroup) {
 	}
 }
 
+// noFilter is a CompactionFilter that does not filter anything
+type noFilter struct {
+}
+
+func (f *noFilter) FilterBlock(key []byte, minTime, maxTime int64) bool {
+	return false
+}
+
+func (f *noFilter) FilterTimeRange(key []byte) (int64, int64) {
```

Comment:
Is it worth moving `tsm1.TimeRange` to the `tsdb` package and returning that value here. By doing so, we could add a concept like `IsZero() bool` or `IsEmpty() bool` that can be used elsewhere, such as:

```go

// Remove any values filtered globally
filterRange := k.compactionFilter.FilterTimeRange(k.blocks[i].key)
if !filterRange.IsZero() {
    v = IntegerValues(v).Exclude(filterRange.Min, filterRange.Max)
    // or better still:
    v = IntegerValues(v).Exclude(filterRange)
}
```

Reply:
Maybe. I'm going to hold off of that change for now.

Reply:
I am happy to take than on later 👍